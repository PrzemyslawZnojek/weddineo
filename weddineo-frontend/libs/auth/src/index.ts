export * from './lib/auth.module';
export * from './lib/+state/auth.facade'
