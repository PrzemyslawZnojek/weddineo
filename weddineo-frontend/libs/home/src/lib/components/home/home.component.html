<mat-card>
  <mat-card-header>
    <h2>{{ 'home.hello' | translate | uppercase }}</h2> </mat-card-header
  ><mat-card-content>
    <h3>{{ 'home.text' | translate }}</h3>

    <h4>{{ 'home.login' | translate }}</h4>

    <weddineo-frontend-action-button
      [width]="'150px'"
      [label]="'auth.login.login' | translate"
      (onClick)="login()"
    ></weddineo-frontend-action-button>

    <h4>{{ 'home.register' | translate }}</h4>

    <weddineo-frontend-action-button
      [width]="'150px'"
      [label]="'auth.register.register' | translate"
      (onClick)="register()"
    ></weddineo-frontend-action-button>
  </mat-card-content>
</mat-card>
