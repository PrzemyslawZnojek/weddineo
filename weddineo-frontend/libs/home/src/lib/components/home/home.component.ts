import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';

@Component({
  selector: 'weddineo-frontend-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
})
export class HomeComponent {
  constructor(private router: Router) {}

  login() {
    this.router.navigateByUrl('auth');
  }

  register() {
    this.router.navigateByUrl('auth/register');
  }
}
