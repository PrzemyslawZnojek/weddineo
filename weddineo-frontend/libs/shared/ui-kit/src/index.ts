export * from './lib/shared-ui-kit.module';
export * from './lib/components/action-button/action-button.component';
export * from './lib/components/text-input/text-input.component';
export * from './lib/validators/email.validator';
export * from './lib/validators/match-fields.validator';
export * from './lib/model/error-message';
export * from './lib/model/select-option';
