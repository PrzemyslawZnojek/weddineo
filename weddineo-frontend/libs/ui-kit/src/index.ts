export * from './lib/ui-kit.module';
export * from './lib/components/action-button/action-button.component';
export * from './lib/components/text-input/text-input.component';
export * from './lib/validators/email.validator';
export * from './lib/validators/match-fields.validator';
