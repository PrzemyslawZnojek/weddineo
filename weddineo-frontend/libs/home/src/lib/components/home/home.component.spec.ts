import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FlexLayoutModule } from '@angular/flex-layout';
import { MockComponents, MockModule, MockPipe } from 'ng-mocks';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatIconModule } from '@angular/material/icon';
import { MatMenuModule } from '@angular/material/menu';

import { TranslatePipe } from '@ngx-translate/core';
import { of } from 'rxjs';
import { AuthFacade } from '@weddineo-frontend/auth';
import { MatButtonModule } from '@angular/material/button';
import { RouterTestingModule } from '@angular/router/testing';
import { HomeComponent } from './home.component';
import { ActionButtonComponent } from '@weddineo-frontend/shared/ui-kit';
import { MatCardModule } from '@angular/material/card';

describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        FlexLayoutModule,
        RouterTestingModule,
        MockModule(MatCardModule),
      ],
      declarations: [ HomeComponent, MockPipe(TranslatePipe), MockComponents(ActionButtonComponent) ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
