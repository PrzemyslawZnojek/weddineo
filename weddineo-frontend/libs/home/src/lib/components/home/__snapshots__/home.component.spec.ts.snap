// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should match snapshot 1`] = `
<weddineo-frontend-home
  router={[Function Router]}
>
  <mat-card>
    <mat-card-header>
      <h2>
        
      </h2>
    </mat-card-header>
    <mat-card-content>
      <h3>
        
      </h3>
      <h4>
        
      </h4>
      <weddineo-frontend-action-button />
      <h4>
        
      </h4>
      <weddineo-frontend-action-button />
    </mat-card-content>
  </mat-card>
</weddineo-frontend-home>
`;
